builtin/providers/aws/structure_test.go

```diff
@@ -824,18 +824,46 @@ func TestFlattenResourceRecords(t *testing.T) {
 			Value: aws.String("127.0.0.1"),
 		},
 		&route53.ResourceRecord{
-			Value: aws.String("127.0.0.3"),
+			Value: aws.String("\"abc def\" \"xxx yyy\""),
 		},
 	}
 
-	result := flattenResourceRecords(expanded)
+	expected_unquoted := []string{
+		"127.0.0.1",
+		"abc def\" \"xxx yyy",
+	}
+	expected_verbatim := []string{
+		"127.0.0.1",
+		"abc def\" \"xxx yyy",
+	}
+
+	checkFlattenResourceRecords(t, "TXT", expanded, expected_unquoted)
+	checkFlattenResourceRecords(t, "SPF", expanded, expected_unquoted)
+
+	checkFlattenResourceRecords(t, "CNAME", expanded, expected_verbatim)
+	checkFlattenResourceRecords(t, "MX", expanded, expected_verbatim)
```

Comment:
This new test is now failing:

```
--- FAIL: TestFlattenResourceRecords (0.00s)
	structure_test.go:865: expected [127.0.0.1 abc def" "xxx yyy], got [127.0.0.1 "abc def" "xxx yyy"]
```

A side note - it would be easier to identify the failing test here if we used named [subtests](https://blog.golang.org/subtests).